### PR TITLE
Assert on responses in tests

### PR DESCRIPTION
**Pull request by gokulchandra**
#17 Adds assertion on responses in tests

### CODE REVIEW

File: src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/AsyncHttpGetDslTest.kt [hunk 1]
```diff
@@ -9,6 +11,14 @@ class AsyncHttpGetDslTest {
 
     @Test
     fun name() {
+        val expectedHeader = hashMapOf(
```
[the remark this answers is not in the document]
Comment (gokulchandra) [reply src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/AsyncHttpGetDslTest.kt]: Nothing specific, I just didn't see any advantage for using a LinkedHashMap there over a HashMap.

File: src/test/kotlin/Utils.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import kotlin.test.assertEquals
+
+
+/**
+ * Created by Gokul on 03/01/2019.
+ */
+
+fun assertResponses(actual: LinkedHashMap<String, Any>, expected: HashMap<String, String>) {
```
[the remark this answers is not in the document]
Comment (gokulchandra) [reply src/test/kotlin/Utils.kt]: The responsibility of `assertResponses` method is to just assert on all the keys received in the response from `postman-echo`. I guess the structure is slightly specific to what `postman-echo` returns. 

Using a Map would be better I guess as it gives better flexibility.

File: src/test/kotlin/Utils.kt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import kotlin.test.assertEquals
```
[the remark this answers is not in the document]
Comment (gokulchandra) [reply src/test/kotlin/Utils.kt]: Done.

File: src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/AsyncHttpGetDslTest.kt [hunk 4]
```diff
@@ -25,7 +35,9 @@ class AsyncHttpGetDslTest {
         }
         runBlocking {
             response.await().use {
-                print(it.body()?.string())
+                val parsedResponse = ObjectMapper().readValue(it.body()?.byteStream(), kotlin.collections.hashMapOf<String, Any>()::class.java)
```
[the remark this answers is not in the document]
Comment (gokulchandra) [reply src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/AsyncHttpGetDslTest.kt]: Good idea. Handling it as a JsonNode is definitely better when the structure is unknown.

File: src/test/kotlin/Utils.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (gokulchandra) [reply src/test/kotlin/Utils.kt]: Regarding point 3, I guess that depends on how we test. 

IMHO I feel that we are testing to see if the parameters, headers etc that we send are actually sent in the Request. 

For example while asserting the headers, there are additional headers like user-agent, host... that we are not really concerned about in that test. So we ignore them while asserting.

File: src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/HttpPostDslKtTest.kt [hunk 5]
```diff
@@ -87,7 +130,12 @@ class HttpPostDslKtTest {
                 }
             }
         }.use {
-            assertEquals(0, it.body()?.string().asJson()["headers"]["content-length"].asInt())
+            val parsedResponse = it.body()?.string().asJson()
+            val headers = parsedResponse["headers"]
+            assertResponses(headers, expectedHeader)
+            assertResponses(parsedResponse["args"], expectedParams)
+            assertEquals("0", headers.get("content-length").asText(""))
```
Comment (rybalkinsd) [inline src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/HttpPostDslKtTest.kt]:
```suggestion
            assertEquals("0", headers.["content-length"].asText(""))
```

File: src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/AsyncHttpGetDslTest.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (rybalkinsd) [reply src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/AsyncHttpGetDslTest.kt]: Ok, I see

File: src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+import com.fasterxml.jackson.databind.JsonNode
+import kotlin.test.assertEquals
+
+/**
+ * Created by Gokul on 03/01/2019.
+ */
+fun assertResponses(actual: JsonNode, expected: Map<String, String>) {
+    expected.forEach { t, u ->
+        assertEquals(u, actual.get(t).asText(""))
```
Comment (rybalkinsd) [inline src/test/kotlin/io/github/rybalkinsd/kohttp/Utils.kt]:
```suggestion
        assertEquals(u, actual[t].asText(""))
```

**Comment by rybalkinsd** [comment]
@gokulchandra congratulations with the first contribution! 🥳 Welcome to join other tasks!

**Pull request merged**